modules/planning/scenarios/park_and_go/stage_adjust.cc

```diff
@@ -49,15 +50,34 @@ Stage::StageStatus ParkAndGoStageAdjust::Process(
   }
   const bool ready_to_cruise =
       scenario::util::CheckADCReadyToCruise(frame, scenario_config_);
-  if (ready_to_cruise) {
-    return FinishStage();
+  const auto vehicle_status = common::VehicleStateProvider::Instance();
+  ADEBUG << vehicle_status->steering_percentage();
+
+  if (!ready_to_cruise) {
+    return StageStatus::RUNNING;
+  }
+  if (std::fabs(vehicle_status->steering_percentage()) <
+      scenario_config_.max_steering_percentage_when_cruise()) {
+    next_stage_ = ScenarioConfig::PARK_AND_GO_CRUISE;
+  } else {
+    next_stage_ = ScenarioConfig::PARK_AND_GO_PRE_CRUISE;
+    // reset init position at current position for pre_cruise stage
+    ResetInitPostion(frame);
```

Comment:
move  to above code (stage transfer. ie. next_stage_ = ?) to FinishStage() to be consistent and clean.

Reply:
Fixed. Thank you.